sources/core/FastSerializer.h

```diff
@@ -449,6 +450,20 @@ inline QString Humanize(FastDeserializer* deserializer)
     return retval;
 }
 
+inline FastDeserializer& operator>>(FastDeserializer& file, std::bitset<4>& bset)
```

Comment:
1. I think, the more suitable file for serialization functions will be `SaveableOperators.h`.
2. If you would like to move this functions to separate file (it will be okay-ish to left them in the `MetalWall.h`) it is needed to make them templates:
```cpp
template<int N>
inline FastDeserializer& operator>>(FastDeserializer& file, std::bitset<N>& bset)
```
because just hardcode size 4 for bitset is kinda weird for a generic serialization function.